src/monitorAnimationSpeed.js

```diff
@@ -0,0 +1,43 @@
+import React, {Component} from 'react'
+import collectFps from 'collect-fps'
+
+const noop = () => {}
+
+export default ({
+  threshold = 30,
+  fpsCollector = collectFps
+}) => (Target) => {
+  let endFPSCollection = noop
+
+  const handleStartFPSCollection = (component) => {
+    if (endFPSCollection !== noop) { endFPSCollection() }
+
+    endFPSCollection = fpsCollector()
+  }
+
+  class MonitorAnimationSpeed extends Component {
+    constructor () {
+      super()
+
+      this.state = {fps: 60}
+    }
+
+    render () {
+      return <Target
+        {...this.props}
+        onStartFPSCollection={handleStartFPSCollection}
+        onEndFPSCollection={() => {
+          if (endFPSCollection !== noop) {
+            this.setState({fps: endFPSCollection()})
+            endFPSCollection = noop
+          }
+        }}
+        lowFPS={this.state.fps < threshold}
+      />
+    }
+  }
+
+  MonitorAnimationSpeed.displayName = Target.displayName || Target.name
```

Comment:
I'm also not sold on the name `Monitor*`, since it can mean basically anything. It could be a component that captures errors from child components and reports them somewhere, for example.

Maybe it's a `RenderPerformanceBenchmarkMonitor` (can you tell I've been writing Java...?)